internal/schema/connection.go

```diff
@@ -3,9 +3,207 @@ package schema
 import (
 	"fmt"
 
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	fwschema "github.com/hashicorp/terraform-plugin-framework/provider/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+func GetEsFWConnectionBlock(keyName string) fwschema.Block {
+	usernamePath := path.MatchRelative().AtParent().AtName("username")
+	passwordPath := path.MatchRelative().AtParent().AtName("password")
+	caFilePath := path.MatchRelative().AtParent().AtName("ca_file")
+	caDataPath := path.MatchRelative().AtParent().AtName("ca_data")
+	certFilePath := path.MatchRelative().AtParent().AtName("cert_file")
+	certDataPath := path.MatchRelative().AtParent().AtName("cert_data")
+	keyFilePath := path.MatchRelative().AtParent().AtName("key_file")
+	keyDataPath := path.MatchRelative().AtParent().AtName("key_data")
+
+	return fwschema.ListNestedBlock{
```

Comment:
Why not use `SingleNestedBlock`?

Reply:
The same question for Kibana and Fleet.

Reply:
There's strict compatibility between the SDK and FW schemas which forces the use of ListNestedBlock here.